### PR TITLE
test: Hibernate jsonb support

### PR DESCRIPTION
**Pull request by pratickchokhani**
1. Support for JSONB added in hibernate.
2. README updated with JSONB support, Hibernate version limitations and PostgreSQL dialect version limitations.

### CODE REVIEW

File: samples/java/hibernate/README.md [hunk 1]
```diff
@@ -50,14 +50,22 @@ Cloud Spanner supports the following data types in combination with `Hibernate`.
 | timestamptz / timestamp with time zone | LocalDateTime     |
 | bytea                                  | byte[]            |
 | date                                   | LocalDate         |
+| jsonb                                  | Custom Data Type  |
+
+### JSONB
+Hibernate 5.* doest not have native support for JSONB. For this, custom data type
```
Comment (olavloite) [inline samples/java/hibernate/README.md]: nit:
```suggestion
Hibernate 5.* does not have native support for JSONB. For this, a custom data type
```

File: samples/java/hibernate/README.md [hunk 2]
```diff
@@ -50,14 +50,22 @@ Cloud Spanner supports the following data types in combination with `Hibernate`.
...
+|----------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Hibernate Version          | Currently, we only support 5.3.x.                                                                                                                                                                                                                                            |
+| PostgreSql Dialect Version | Currently, we only support 42.4.3.                                                                                                                                                                                                                                           |
```
Comment (olavloite) [inline samples/java/hibernate/README.md]: Should this be:
```suggestion
| PostgreSql JDBC driver Version | Currently, we only support 42.4.3.                                                                                                                                                                                                                                           |
```
?

File: samples/java/hibernate/README.md [hunk 3]
```diff
@@ -50,14 +50,22 @@ Cloud Spanner supports the following data types in combination with `Hibernate`.
...
-| Generated primary keys               | Cloud Spanner does not support `sequences`. Auto-increment primary key is not supported. Remove auto increment annotation for primary key columns. The recommended type of primary key is a client side generated `UUID` stored as a string. |
+| Limitation                 | Workaround                                                                                                                                                                                                                                                                   |
+|----------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Hibernate Version          | Currently, we only support 5.3.x.                                                                                                                                                                                                                                            |
```
Comment (olavloite) [inline samples/java/hibernate/README.md]: I think we should make this a little simpler / less suggestive (the current text sounds like we're actively working on other versions)

```suggestion
| Hibernate Version          | Version 5.3.x. is supported                                                                                                                                                                                                                                           |
```

Do we know that it breaks with other versions? Or is this just a safeguard?

File: samples/java/hibernate/src/main/java/com/google/cloud/postgres/models/VenueDescription.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package com.google.cloud.postgres.models;
```
Comment (olavloite) [inline samples/java/hibernate/src/main/java/com/google/cloud/postgres/models/VenueDescription.java:14]: nit: add copyright header

File: samples/java/hibernate/src/main/java/com/google/cloud/postgres/models/Venues.java [hunk 5]
```diff
@@ -39,8 +40,9 @@ public class Venues {
   @Column(name = "name", nullable = false)
   private String name;
 
-  @Column(name = "description", nullable = false)
-  private String description;
+  @Column(name = "description", nullable = false, columnDefinition = "jsonb")
```
Comment (olavloite) [inline samples/java/hibernate/src/main/java/com/google/cloud/postgres/models/Venues.java:43]: Has the script that creates the sample data model been updated? Or is that not needed?

File: samples/java/hibernate/README.md (hunk 3 above)
Comment (pratickchokhani) [reply samples/java/hibernate/README.md]: For hibernate, this is just a safe guard but for Postgresql driver we know that it doesn't work with v42.5.0.

File: samples/java/hibernate/src/main/java/com/google/cloud/postgres/models/Venues.java (hunk 5 above)
Comment (pratickchokhani) [reply samples/java/hibernate/src/main/java/com/google/cloud/postgres/models/Venues.java:43]: Done.

File: samples/java/hibernate/README.md (hunk 1 above)
Comment (pratickchokhani) [reply samples/java/hibernate/README.md]: Done.

File: samples/java/hibernate/README.md (hunk 2 above)
Comment (pratickchokhani) [reply samples/java/hibernate/README.md]: Done

File: samples/java/hibernate/src/main/java/com/google/cloud/postgres/models/VenueDescription.java (hunk 4 above)
Comment (pratickchokhani) [reply samples/java/hibernate/src/main/java/com/google/cloud/postgres/models/VenueDescription.java:14]: Done

File: samples/java/hibernate/README.md (hunk 3 above)
Comment (olavloite) [reply samples/java/hibernate/README.md]: Would you mind sharing (offline) the SQL statement that 42.5.0 generates that fails?

**Review by olavloite — APPROVED** [review]

**Pull request merged**